### ISSUE TITLE
Tracking

### ISSUE THREAD

**Issue by tschubotz** [gnosis/safe#299]
# What is this feature about? (1 sentence)
This will allow us to measure the use and success of our app.

# Why is it needed? What is the value? For whom do we build it?
There is no way to manage product without measuring success. This helps us determine areas for improvement.

# High-level overview of the feature
- Set up Firebase analytics: https://firebase.google.com/docs/analytics
  - I suggest using a new firebase projects to separate this from the legacy app
- Make sure that default [user properties](https://support.google.com/firebase/answer/6317486?hl=en) 
- Please set the following 2 [custom user properties](https://firebase.google.com/docs/analytics/user-properties?platform=ios):
  - `num_safes`
    - Integer (Edit June 15: Firebase only allows to set strings, which is fine.)
    - Should be the number of Safes that the user has added on their device.
    - Please make sure that this number is 0 after fresh install and that it is increased when adding a Safe and decreased when removing a Safe from the device.
  - ~`push_info`~ (Edit June 15 remove from this epic since push notifications will only be added after the public release.)
    - ~string~
    - ~`unknown` if we haven't asked for push access yet (iOS only since we have push permission on Android by default)~
    - ~`enabled` if we have push notification permission (or enabled it after previously denying)~
    - ~`disabled` if the user denied push permission (or removed it after previously allowing)~
- [Personalized advertising features are disabled](https://firebase.google.com/docs/analytics/configure-data-collection?platform=ios#disable_personalized_advertising_features)
- (Edit June 15) Please track the screens as separate events with `screen_` prefix according to  https://github.com/gnosis/safe/wiki/Tracking
- (Edit June 29) This Epic is only for screen tracking of the following screens. (Other screens are part of the respective Epics)
  - `launch`
  - `launch_terms`
  - `assets_no_safe`
  - `assets_coins`
  - `assets_collectibles`
  - `safe_add_address`
  - `safe_add_name`
  - `safe_add_ens`
  - `camera`
  - `safe_receive`
  - `safe_switch`

Tracking info is replicated in [wiki](https://github.com/gnosis/safe/wiki/Tracking)

# Updates
- June 29: Limit this ticket to screens of epic #285, #286, and #292. Tracking of other screens will be part of respective ticket.

**Comment by rmeissner** [comment]
`push_info` what is expected for that on Android?

**Comment by tschubotz** [comment]
> `push_info` what is expected for that on Android?

Assuming it is possible to get the info whether the user blocked all pushes (https://gnosisinc.slack.com/archives/C63LQTGNB/p1586183378023000), I expect that on Android `push_info` is `enabled` by default and switches to `disabled` if the user blocked all pushes.

**Comment by jpalvarezl** [comment]
In this situation, I would say then that we would have to have only a single notification channel. Alternatively, if we have many, we could return the channel name that was disabled as a parameter.

**Comment by elgatovital** [comment]
should we mention tracking in privacy policy (gdpr compliancy)

**Comment by tschubotz** [comment]
> should we mention tracking in privacy policy (gdpr compliancy)

Yes it's already mentioned under `4.5.   Firebase` of our gnosis-safe.io/privacy/ policy already that we do tracking.

`num_safe` and `push_info` are not PII in my opinion but will double check again.

**Comment by sche** [comment]
I think we should have an option to disable tracking from app settings.

**Comment by sche** [comment]
I remember we had some issues with automatic screen tracking, so we decided to use explicit tracking in the legacy app. We will need to revisit it, but we should keep it in mind.

**Comment by tschubotz** [comment]
> I think we should have an option to disable tracking from app settings.

Tracking is crucial for us in the beginning to measure success. I agree though that this would be nice. Let's add it at some point in the future. Also I haven't heard more than a handful complains from users about the fact that we do tracking. 

> I remember we had some issues with automatic screen tracking, so we decided to use explicit tracking in the legacy app. We will need to revisit it, but we should keep it in mind.

Yes, if those issues still persist, let's talk about the workaround again.

**Comment by sche** [comment]
@tschubotz this story depends on #290 which is after the public release. Would you mind to create a separate story for "push_info" so we could close this story sooner?

**Comment by tschubotz** [comment]
totally, good point, will do!

**Comment by sche** [comment]
@tschubotz user properties values can only be strings:
https://firebase.google.com/docs/reference/swift/firebaseanalytics/api/reference/Classes/Analytics#setuserproperty_:forname:
https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#setUserProperty(java.lang.String,%20java.lang.String)

**Comment by tschubotz** [comment]
That's fine. I think Vitaly or Richard brought this up already. Firebase seems to handle this properly when working with them in the UI: https://stackoverflow.com/questions/46281733/how-to-send-numbers-in-user-attributes-on-firebase :) 
I'll add to the epic + wiki that they are strings only to make sure we don't forget.

**Comment by liliya-soroka** [comment]
Verified
The events listed in the Epic are tracked in both apps
